Source/RiveEditor/Private/RiveFile_AssetTypeActions.cpp

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright Rive, Inc. All rights reserved.
+
+
+#include "RiveFile_AssetTypeActions.h"
+
+#include "Interfaces/ITextureEditorModule.h"
+#include "Interfaces/ITextureEditorModule.h"
+#include "Rive/RiveFile.h"
+
+#define LOCTEXT_NAMESPACE "RiveEditorModule"
+
+FRiveFile_AssetTypeActions::FRiveFile_AssetTypeActions(EAssetTypeCategories::Type InAssetCategory)
+	: AssetCategory(InAssetCategory)
+{
+}
+
+FText FRiveFile_AssetTypeActions::GetName() const
+{
+	return LOCTEXT("FRiveFile_AssetTypeActions_Name", "Rive File");
+}
+
+FColor FRiveFile_AssetTypeActions::GetTypeColor() const
+{
+	return FColor::Red;
+}
+
+UClass* FRiveFile_AssetTypeActions::GetSupportedClass() const
+{
+	return URiveFile::StaticClass();
+}
+
+FText FRiveFile_AssetTypeActions::GetAssetDescription(const FAssetData& AssetData) const
+{
+	return FAssetTypeActions_Base::GetAssetDescription(AssetData);
+}
+
+bool FRiveFile_AssetTypeActions::IsImportedAsset() const
+{
+	return true;
+}
+
+void FRiveFile_AssetTypeActions::OpenAssetEditor(const TArray<UObject*>& InObjects, TSharedPtr<IToolkitHost> EditWithinLevelEditor)
+{
+	if (ensure(InObjects.Num() == 1) && InObjects[0]->IsA<URiveFile>())
+	{
+		ITextureEditorModule* TextureEditorModule = &FModuleManager::LoadModuleChecked<ITextureEditorModule>("TextureEditor");
+		TextureEditorModule->CreateTextureEditor(EToolkitMode::Standalone, EditWithinLevelEditor, Cast<URiveFile>(InObjects[0]));
```

Comment:
We have a custom editor where we set rive file artboard, resolution, etc. How that would work

Reply:
```
EAssetCommandResult URiveFile_AssetDefinitionDefault::OpenAssets(const FAssetOpenArgs& OpenArgs) const
{
	for (URiveFile* RiveFile : OpenArgs.LoadObjects<URiveFile>())
	{
		const TSharedRef<FRiveAssetToolkit> EditorToolkit = MakeShared<FRiveAssetToolkit>();
		EditorToolkit->Initialize(RiveFile, OpenArgs.GetToolkitMode(), OpenArgs.ToolkitHost);
	}

	return EAssetCommandResult::Handled;
}

```

Reply:
Thanks, was looking for this